ui/src/shared/components/Dygraph.js

```diff
@@ -45,13 +45,18 @@ class Dygraph extends Component {
       isGraphFilled: fillGraph,
       isBarGraph,
       options,
+      labels,
     } = this.props
 
     const timeSeries = this.timeSeries
     const graphRef = this.graphRef
 
     let defaultOptions = {
+      ...options,
+      labels,
       fillGraph,
+      file: this.timeSeries,
```

Comment:
looks like this only applies on `updateOptions`. how does this help fix the bug?

Reply:
:thu